tools/network/dnssec/sort_test.go

```diff
@@ -32,14 +32,26 @@ func TestSrvSort(t *testing.T) {
 	arr := make([]*net.SRV, 0, 7)
 	arr = append(arr, &net.SRV{Priority: 4, Weight: 1})
 	arr = append(arr, &net.SRV{Priority: 3, Weight: 1})
-	arr = append(arr, &net.SRV{Priority: 1, Weight: 200})
+	arr = append(arr, &net.SRV{Priority: 1, Weight: 0xFFFF}) // max possible value to increase the ordering probability
 	arr = append(arr, &net.SRV{Priority: 1, Weight: 1})
 	arr = append(arr, &net.SRV{Priority: 1, Weight: 1})
 	arr = append(arr, &net.SRV{Priority: 1, Weight: 1})
 	arr = append(arr, &net.SRV{Priority: 1, Weight: 1})
 
+	retryCounter := 0
+retry:
 	srvRecArray(arr).sortAndRand()
-	a.Equal(net.SRV{Priority: 1, Weight: 200}, *arr[0])
+	if (*arr[0] != net.SRV{Priority: 1, Weight: 0xFFFF}) {
+		// there is a small change that a random number from 0 to max uint15 would be 0 or 1
```

Comment:
I'm going to approve in the interest of fixing tests, but why not just limit the range of the random number to be greater than 1?

Reply:
the spec requires to pick a random number from 0 to sum of all weights to be ordered... I guess it is better to be more-less spec compliant.